### PR TITLE
feat: made it so a level can now be finished + deleted some magic num…

### PR DESCRIPTION
**Pull request by waseland**
…bers

📒 🚔 🔴 🌮 😂 🍏 🍌 🍆

### CODE REVIEW

File: src/gamelogic/Game.cpp [hunk 1]
```diff
@@ -99,5 +100,15 @@ namespace GameLogic {
...
     }
+
+    SPTR_Level Game::get_level() {
+        if(this->_level != nullptr){
+            return this->_level;
+        }
+        else {
+            throw Exceptions::LevelIsNullptrException();
+        }
+
```
Comment (jorgdebont) [inline src/gamelogic/Game.cpp]: empty line :O

File: src/gamelogic/state/Menu.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ namespace State {
         }
 
         this->_context.SDL_facade.draw_rect(
-            {120, this->get_selected()->coordinates.y + 6}, //todo delete this magic number
+            {this->get_selected()->coordinates.x - this->_selected_rect_diff_width, this->get_selected()->coordinates.y + this->_selected_rect_diff_height},
             15,
             15,
             _color
```
Comment (jorgdebont) [inline src/gamelogic/state/Menu.cpp]: this :D

File: src/gamelogic/Game.cpp [hunk 3]
```diff
@@ -99,5 +100,15 @@ namespace GameLogic {
         this->SDL_facade.load_sound_effect("monsterkill", VIDYA_RUNPATH + "res/sound_effects/monsterkill.wav");
         this->SDL_facade.load_sound_effect("headshot", VIDYA_RUNPATH + "res/sound_effects/headshot.wav");
     }
+
+    SPTR_Level Game::get_level() {
+        if(this->_level != nullptr){
```
Comment (jorgdebont) [inline src/gamelogic/Game.cpp]: no spaces between :-1:

File: src/gamelogic/Level.cpp [hunk 4]
```diff
@@ -98,4 +98,13 @@ namespace GameLogic {
     {
         return this->_field;
     }
+
+    void Level::set_level_over()
+    {
+        this->_level_won = true;
+    }
+
+    bool Level::is_level_over() {
```
Comment (jorgdebont) [inline src/gamelogic/Level.cpp]: { not on new line

File: src/gamelogic/exceptions/LevelIsNullptrException.cpp [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#include "LevelIsNullptrException.hpp"
+
+namespace Exceptions {
+    const char * LevelIsNullptrException::what() const throw( )
+    {
+        return "Game._level was a nullptr";
```
Comment (jorgdebont) [inline src/gamelogic/exceptions/LevelIsNullptrException.cpp]: Why is game._level assumed, name suggest that is just a level null ptr

File: src/gamelogic/Game.cpp (hunk 3 above)
Comment (waseland) [inline src/gamelogic/Game.cpp]: A

File: src/gamelogic/Game.cpp (hunk 1 above)
Comment (waseland) [inline src/gamelogic/Game.cpp]: B

File: src/gamelogic/Game.cpp [hunk 6]
```diff
@@ -99,5 +100,13 @@ namespace GameLogic {
         this->SDL_facade.load_sound_effect("monsterkill", VIDYA_RUNPATH + "res/sound_effects/monsterkill.wav");
         this->SDL_facade.load_sound_effect("headshot", VIDYA_RUNPATH + "res/sound_effects/headshot.wav");
     }
+
+    SPTR_Level Game::get_level() {
+        if (this->_level != nullptr){
```
Comment (jorgdebont) [inline src/gamelogic/Game.cpp]: fuck you

**Pull request merged**